presto-hana/src/main/java/com/facebook/presto/plugin/hana/HanaClient.java

```diff
@@ -0,0 +1,142 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.hana;
+
+import com.facebook.presto.plugin.jdbc.BaseJdbcClient;
+import com.facebook.presto.plugin.jdbc.BaseJdbcConfig;
+import com.facebook.presto.plugin.jdbc.ConnectionFactory;
+import com.facebook.presto.plugin.jdbc.DriverConnectionFactory;
+import com.facebook.presto.plugin.jdbc.JdbcColumnHandle;
+import com.facebook.presto.plugin.jdbc.JdbcConnectorId;
+import com.facebook.presto.plugin.jdbc.JdbcIdentity;
+import com.facebook.presto.plugin.jdbc.JdbcTableHandle;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.SchemaTableName;
+import com.google.common.base.Joiner;
+import com.sap.db.jdbc.Driver;
+
+import javax.inject.Inject;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+
+import static com.facebook.presto.plugin.jdbc.DriverConnectionFactory.basicConnectionProperties;
+import static com.facebook.presto.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+
+public class HanaClient
+        extends BaseJdbcClient
+{
+    private static final Joiner DOT_JOINER = Joiner.on(".");
+
+    @Inject
+    public HanaClient(JdbcConnectorId connectorId, BaseJdbcConfig config)
+            throws SQLException
+    {
+        super(connectorId, config, "\"", connectionFactory(config));
+    }
+
+    private static ConnectionFactory connectionFactory(BaseJdbcConfig config)
+            throws SQLException
+    {
+        Properties connectionProperties = basicConnectionProperties(config);
+        return new DriverConnectionFactory(
+                new Driver(),
+                config.getConnectionUrl(),
+                Optional.ofNullable(config.getUserCredentialName()),
+                Optional.ofNullable(config.getPasswordCredentialName()),
+                connectionProperties);
+    }
+
+    @Override
+    public void abortReadConnection(Connection connection)
+            throws SQLException
+    {
+        // Abort connection before closing. Without this, the Hana driver
+        // attempts to drain the connection by reading all the results.
+        connection.abort(directExecutor());
```

Comment:
It would be better to use try catch here

Reply:
Thank you Shengxuan for the review!  @JiamingMai do you mind to take a look?

Reply:
Updated. Thanks @liushengxuan. I used try catch here.